From 813aaaca4fc3e3beed3d1bbd2970149c397fb9b5 Mon Sep 17 00:00:00 2001
From: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
Date: Wed, 5 Feb 2025 16:24:20 -0800
Subject: [PATCH] QSDK:CP: Add compile flag for ATH12K Host Test Framework for
 11s MLO

This adds a compile time flag to enable ATH12K Host Test Framework
for 11s MLO. As the changes are just meant to test firmware support for
11s MLO, this flag is not by default enabled. User might enable
this using make menuconfig.

Change-Id: Ifae0e243a0516d14eb939ab4a8d3e63cbf06f268
Signed-off-by: Pradeep Kumar Chitrapu <quic_pradeepc@quicinc.com>
---
 package/kernel/mac80211/ath.mk | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/package/kernel/mac80211/ath.mk b/package/kernel/mac80211/ath.mk
index b7402518518b..2d0fead16f29 100644
--- a/package/kernel/mac80211/ath.mk
+++ b/package/kernel/mac80211/ath.mk
@@ -77,6 +77,7 @@ config-$(call config_package,ath12k) += ATH12K
 
 config-$(CONFIG_PACKAGE_kmod-ath12k) += ATH12K_SPECTRAL
 config-$(CONFIG_PACKAGE_ATH12K_SAWF) += ATH12K_SAWF
+config-$(CONFIG_PACKAGE_ATH12K_TEST_FW_FOR_11S_MLO) += ATH12K_TEST_FW_FOR_11S_MLO
 ifeq ($(CONFIG_KERNEL_IPQ_MEM_PROFILE),512)
 config-y += ATH12K_MEM_PROFILE_512M
 endif
@@ -356,6 +357,13 @@ define KernelPackage/ath12k/config
 		help
 			This option enables support for SAWF and Telemetry
 			in ATH12K.
+
+	config PACKAGE_ATH12K_TEST_FW_FOR_11S_MLO
+		bool "Enable 11S MLO host test Framework in ATH12K"
+		default n
+		help
+			This option enables support for 11S MLO Host Test Framework
+			in ATH12K.
 endef
 
 define KernelPackage/ath11k-ahb
-- 
2.34.1

